sockeye/inference.py

```diff
@@ -661,43 +661,40 @@ def translate(self, trans_inputs: List[TranslatorInput]) -> List[TranslatorOutpu
                 input_chunks.append(trans_input.tokens)
                 chunk_markers.append(trans_input.id)
 
-        # start translation in batches
+        # translate in batch-sized blocks over input chunks
         translations = []  # type: List[Translation]
-        for i in range(0, len(input_chunks), self.batch_size):
-            logger.debug("Translating batch %d", i)
-            batch = input_chunks[i:i + self.batch_size]
+        for batch_id, batch in enumerate(utils.grouper(input_chunks, self.batch_size)):
+            logger.debug("Translating batch %d", batch_id)
             # underfilled batch will be filled to a full batch size with copies of the 1st input
             rest = self.batch_size - len(batch)
             if rest > 0:
                 logger.debug("Extending the last batch to the full batch size (%d)", self.batch_size)
                 batch = batch + [batch[0]] * rest
-            # translate
             translated_batch = self.translate_nd(*self._get_inference_input(batch))
             # truncate to remove duplicated translations
             if rest > 0:
                 translated_batch = translated_batch[:-rest]
             translations.extend(translated_batch)
 
-        # concatenate translations corresponding to spans of chunk markers with the same input id
+        # concatenate translations with the same input id into a dict of complete translations
+        concat_translations = dict()
```

Comment:
you might need a type annotation here to make mypy happy

Reply:
looks like it's happy with it already:)